src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -22,23 +22,32 @@ public abstract class DesktopBuildClientTests : TestBase
     {
         private sealed class TestableDesktopBuildClient : DesktopBuildClient
         {
+            /// <summary>
+            /// Stores fully constructed pipe name.
+            /// </summary>
             private readonly string _pipeName;
+            /// <summary>
+            /// Stores unhashed "uniqueifying" pipe name component.
```

Comment:
It seems like using the shared compilation id as a component, as opposed to either using it verbatim or hashing it and using that value verbatim, is unnecessarily complicated. Is there a scenario you're thinking of where this is necessary?